src/nomad_measurements/xrd/xrd_parser.py

```diff
@@ -208,10 +221,66 @@ def parse_and_convert_file(file_path):
     '''
     file_path = os.path.abspath(file_path)
 
-    format_parser = FormatParser(file_path)
+    format_parser = FormatParser(file_path, logger)
     parsed_data = format_parser.parse()
 
     data_converter = DataConverter(parsed_data)
     common_data = data_converter.convert()
 
     return common_data
+
+def read_rigaku_RASX(file_path, logger):
+    ''' Reads .rasx files from Rigaku instruments
+        - reader is based on IKZ submodule 
+        - currently supports one scan per file
+        - in case of multiple scans per file, only the first scan is read
+    Args:

```

Comment:
Ah, and also start the docstring with an empty line. Remind me later and I'll help you setup the vscode doctring extension so that it agrees with the NOMAD standard :)